plugins/dql-backend/src/service/router.ts

```diff
@@ -53,6 +54,20 @@ export const createRouter = async (
   const router = Router();
   router.use(express.json());
 
+  router.get('/catalog/:queryId', async (req, res) => {
```

Comment:
So at the moment there is a call for every query inside the component in `catalog-info.yaml`.
I'm not sure if this will lead to any side effects, except when the user has some rate-limits in place for their backstage instance.
@MrManny @TorbenGallobDynatrace your opinion on that?

Reply:
Definitely the better solution.

The backend should return the results in a nested array, as the backend already knows which queries the component has.